bmcs/src/main/scala/akka/stream/alpakka/oracle/bmcs/javadsl/BmcsClient.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.oracle.bmcs.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import scala.compat.java8.FutureConverters._
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.javadsl.model.headers.ByteRange
+import akka.http.javadsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.{ByteRange => ScalaByteRange}
+import akka.stream.Materializer
+import akka.stream.alpakka.oracle.bmcs.BmcsSettings
+import akka.stream.alpakka.oracle.bmcs.auth.BmcsCredentials
+import akka.stream.alpakka.oracle.bmcs.impl.{BmcsStream, CompleteMultipartUploadResult, ObjectSummary}
+import akka.stream.javadsl.{Sink, Source}
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+
+final case class MultipartUploadResult(bucketName: String, objectName: String, etag: String)
```

Comment:
Often we have put model classes/settings that are shared between java and scala in a single package one level up, which would be `akka.stream.alpakka.oracle.bmcs` in this case. For those we have to make sure to provide both Scala and Java style constructors, methods and types used.

For types that are just returned it can make sense to represent the return type with a trait/interface instead as having a case class gives a lot of methods that then becomes public api.